patterns/structural/3-tier.py

```diff
@@ -3,68 +3,73 @@
 Separates presentation, application processing, and data management functions.
 """
 
+from typing import Dict, KeysView, Optional, Type, TypeVar, Union
+
 
 class Data:
     """ Data Store Class """
 
     products = {
-        'milk': {'price': 1.50, 'quantity': 10},
-        'eggs': {'price': 0.20, 'quantity': 100},
-        'cheese': {'price': 2.00, 'quantity': 10},
+        "milk": {"price": 1.50, "quantity": 10},
+        "eggs": {"price": 0.20, "quantity": 100},
+        "cheese": {"price": 2.00, "quantity": 10},
     }
 
     def __get__(self, obj, klas):
+
         print("(Fetching from Data Store)")
-        return {'products': self.products}
+        return {"products": self.products}
 
 
 class BusinessLogic:
     """ Business logic holding data store instances """
 
     data = Data()
 
-    def product_list(self):
-        return self.data['products'].keys()
+    def product_list(self) -> KeysView[str]:
+        return self.data["products"].keys()
 
-    def product_information(self, product):
-        return self.data['products'].get(product, None)
+    def product_information(
+        self, product: str
+    ) -> Optional[Dict[str, Union[int, float]]]:
+        return self.data["products"].get(product, None)
 
 
 class Ui:
     """ UI interaction class """
 
-    def __init__(self):
+    def __init__(self) -> None:
```

Comment:
My only objection is with the `None` returns. Can you give me examples of other projects that use it? I don't see an added value.

Reply:
I think its part of the convention [here is the discussion](https://github.com/python/typing/issues/65#issuecomment-96904050)

Reply:
Here's an example from `Black` formatter's [source code.](https://github.com/psf/black/blob/2471b9256d9d9dfea1124d20072201693b9b0865/src/black/__init__.py#L119)